pipe.go

```diff
@@ -231,15 +232,16 @@ func makeServerPipeHandle(path string, securityDescriptor []byte, c *PipeConfig,
 		mode |= cPIPE_TYPE_MESSAGE
 	}
 
-	var sa securityAttributes
-	sa.Length = uint32(unsafe.Sizeof(sa))
+	sa := &securityAttributes{}
+	sa.Length = uint32(unsafe.Sizeof(*sa))
 	if securityDescriptor != nil {
 		sa.SecurityDescriptor = &securityDescriptor[0]
```

Comment:
>&securityDescriptor[0] [](start = 26, length = 22)

This actually violates cgo rules: [https://golang.org/cmd/cgo/](https://golang.org/cmd/cgo/)

Probably we should be duplicating this into a C-allocated buffer.